Modules/Devices/src/Devices.Application/Identities/Commands/CreateIdentity/Handler.cs

```diff
@@ -46,9 +47,10 @@ public async Task<CreateIdentityResponse> Handle(CreateIdentityCommand command,
         if (existingIdentity != null)
             throw new OperationFailedException(ApplicationErrors.Devices.AddressAlreadyExists());
 
-        var basicTier = await _tiersRepository.GetBasicTierAsync(cancellationToken);
+        var tierId = TierId.Create(command.TierId).Value;
+        var tier = await _tiersRepository.FindById(tierId, cancellationToken) ?? throw new OperationFailedException(ApplicationErrors.Devices.InvalidTierId());
```

Comment:
Instead of using the passed id, you should use the default tier that was configured for the client. Why else would we have a default tier? 😮